Trimble.Modus.Components/Controls/Button/CustomButton.xaml.cs

```diff
@@ -110,174 +122,234 @@ public new Color BackgroundColor
+                UpdateOutlineStyleColors(customButton);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private static void SetPadding(CustomButton customButton)
+    {
+        if (!customButton.IsFloatingButton)
+        {
+            switch (customButton.Size)
+            {
+                case Enums.Size.XSmall:
+                    customButton.buttonLabel.FontSize = (double)Enums.FontSize.XSmall;
+                    customButton.buttonStackLayout.Padding = new Thickness(16, 6);
+                    customButton.HeightRequest = 32;
+                    break;
+                case Enums.Size.Small:
+                    customButton.buttonLabel.FontSize = (double)Enums.FontSize.Small;
+                    customButton.buttonStackLayout.Padding = new Thickness(16, 8);
+                    customButton.HeightRequest = 40;
+                    break;
+                case Enums.Size.Default:
+                    customButton.buttonLabel.FontSize = (double)Enums.FontSize.Default;
+                    customButton.buttonStackLayout.Padding = new Thickness(16, 12);
+                    customButton.HeightRequest = 48;
+                    break;
+                case Enums.Size.Large:
+                    customButton.buttonLabel.FontSize = (double)Enums.FontSize.Large;
+                    customButton.buttonStackLayout.Padding = new Thickness(16, 12);
+                    customButton.HeightRequest = 56;
+                    break;
+                default:
+                    break;
+
+            }
+        }
+        else
+        {
+            if (customButton.Title != null)
+            {
+                customButton.buttonStackLayout.Padding = new Thickness(16, 0);
+            }
+            else
+            {
+                customButton.buttonStackLayout.Padding = new Thickness(16);
+            }
+        }
+
+    }
+
+    private static void UpdateFillStyleColors(CustomButton customButton)
+    {
+        switch (customButton.ButtonColor)
+        {
+            case ButtonColor.Secondary:
+                customButton.buttonFrame.BackgroundColor = ResourcesDictionary.ColorsDictionary("SecondaryButton");
+                customButton.buttonFrame.Stroke = ResourcesDictionary.ColorsDictionary("TrimbleBlue");
+                customButton.buttonLabel.TextColor = ResourcesDictionary.ColorsDictionary("White");
+                customButton.tempColor = customButton.buttonFrame.BackgroundColor;
+
+                break;
+
+            case ButtonColor.Tertiary:
+                customButton.buttonFrame.BackgroundColor = ResourcesDictionary.ColorsDictionary("TertiaryButton");
+                customButton.buttonFrame.Stroke = ResourcesDictionary.ColorsDictionary("Transparent");
+                customButton.buttonLabel.TextColor = ResourcesDictionary.ColorsDictionary("TrimbleGray");
+                customButton.tempColor = customButton.buttonFrame.BackgroundColor;
```

Comment:
Why templcolor required?

Reply:
Its required to set the color of that particular style (fill,outline,borderless) so after setting the color of the background while clicking it ,we are s betting the color back to the previous color of the background

Reply:
Can you rename the variable? like 'ActiveColor'?

Reply:
Sure